### PR TITLE
[Station Objective] Drone Dispenser

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

Adds a new station objective; Drone Dispenser. This needs machine board bought with cargo, the parts and enough mats to produce N number of drone shells.

## Why It's Good For The Game

People have been enjoying drone roles for a while. This may be a good idea. 

Maybe I should add the option to disable it in config (LRP), but I'm not sure.

## Changelog
:cl:
add: Added Drone Dispenser Station Objective
tweak: Drone Dispensers track how many drones they produced
/:cl:

### CODE REVIEW

**Comment by williamcll** [comment]
Would be great if the drone crate can be opened by RD or roboticist.

**Comment by eLeCtrOssSnake** [comment]
Seems like a cool idea.

> Would be great if the drone crate can be opened by RD or roboticist.

Why not?

**Comment by eLeCtrOssSnake** [comment]
>Maybe I should add the option to disable it in config (LRP), but I'm not sure.

Why remove it from LRP?

**Comment by eeSPee** [comment]
> Maybe I should add the option to disable it in config (LRP), but I'm not sure.
> Why remove it from LRP?

Minimizer griff?

**Comment by eLeCtrOssSnake** [comment]
If it allows griefing in any way, that means it must be fixed before being merged. If it introduces exploits/griefs/lags there is no difference on LRP/MRP in a possibility of a game mechanic abuse by some player, especially if it will ruin other player’s games too.

**Comment by SergeiKoralev** [comment]
As an admin: Please for the love of god no.

If you're going to follow through on this please make it a config option.

File: code/modules/station_goals/drone_dispenser.dm [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	drones_produced = rand(1,5)
+
+/datum/station_goal/drone_dispenser/get_report()
+	return {"We want you to construct the requirements for a self-sustaining, autonomous research facility.
+	 This would require you to construct a Drone Dispenser aboard your station.
+
+	 The Drone Dispenser needs to be operational at the end of the shift,
+	 and must have dispensed at least [drones_produced] drone shells.
+
+	 Base vault parts are available for shipping via cargo."}
```
Comment (ghost) [inline code/modules/station_goals/drone_dispenser.dm]: Copy-paste from DNA vault? Regardless,
```suggestion
	 Dispenser circuitry are available for shipping via cargo."}
```

**Pull request closed without merging**